clarifai/client/lister.py

```diff
@@ -0,0 +1,88 @@
+from datetime import datetime
+from typing import Any, Callable, Dict, Generator
+
+from clarifai_grpc.grpc.api.status import status_code_pb2
+from google.protobuf.json_format import MessageToDict
+from google.protobuf.timestamp_pb2 import Timestamp
+
+from clarifai.client.base_client import BaseClient
+
+
+class Lister(BaseClient):
+
+  def __init__(self, page_size: int = 16):
+    self.default_page_size = page_size
+
+  def convert_string_to_timestamp(self, date_str) -> Timestamp:
+    """Converts a string to a Timestamp object.
+    Args:
+        date_str (str): The string to convert.
+    Returns:
+        Timestamp: The converted Timestamp object.
+    """
+    # Parse the string into a Python datetime object
+    try:
+      datetime_obj = datetime.strptime(date_str, '%Y-%m-%dT%H:%M:%S.%fZ')
+    except ValueError:
+      datetime_obj = datetime.strptime(date_str, '%Y-%m-%dT%H:%M:%SZ')
+
+    # Convert the datetime object to a Timestamp object
+    timestamp_obj = Timestamp()
+    timestamp_obj.FromDatetime(datetime_obj)
+
+    return timestamp_obj
+
+  def convert_keys_to_snake_case(self, old_dict: Dict[str, Any],
```

Comment:
Why is this needed?
I would think the integrity of the data fetched from the APIs must be maintained.

Reply:
This is needed for initialising/creating objects from respective resource protos.

The MultiOutputResponse returns response with keys as `userId`, `defaultLanguage` etc... but the protos expects all keys to be snake cased user_id, default_language

Reply:
It looks like you do the convertion after retrieving data from the `List` endpoints

Reply:
Yes, we get all responses from `List` endpoints and return as respective class object for each of the response

Reply:
Yes. Only the keys are modified, not the values

Reply:
Got it